### PR TITLE
Add X509Chain tests for certificate expiry

### LINKED ISSUE #3028: Add X509Chain tests for expiration boundary conditions
The X509Chain object supports an explicit time for the override.

Using a statically built chain, no revocation mode, and an explicit time, it should be possible to get a very deterministic test for the chain assessment.

### PR DESCRIPTION
**Pull request by bartonjs**
Fixes #3028.

cc: @morganbr @AtsushiKan

### CODE REVIEW

File: src/System.Security.Cryptography.X509Certificates/tests/ChainTests.cs [hunk 1]
```diff
@@ -65,5 +66,92 @@ public static void BuildChainExtraStoreUntrustedRoot()
...
+            }
+
+            return testCases;
+        }
+
+        [Theory]
+        [MemberData("VerifyExpressionData")]
+        public static void VerifyExpiration_LocalTime(DateTime verificationTime, bool shouldBeValid, DateTimeKind kind)
+        {
+            using (var microsoftDotCom = new X509Certificate2(TestData.MicrosoftDotComSslCertBytes))
```
Comment (morganbr) [inline src/System.Security.Cryptography.X509Certificates/tests/ChainTests.cs]: I don't see this anywhere. Did you forget a file?

File: src/System.Security.Cryptography.X509Certificates/tests/ChainTests.cs [hunk 2]
```diff
@@ -65,5 +66,92 @@ public static void BuildChainExtraStoreUntrustedRoot()
                 Assert.Contains(chain.ChainStatus, s => s.Status == X509ChainStatusFlags.UntrustedRoot);
             }
         }
+
+        public static IEnumerable<object[]> VerifyExpressionData()
+        {
+            // The test will be using the chain for TestData.MicrosoftDotComSslCertBytes
+            // The leaf cert (microsoft.com) is valid from 2014-10-15 00:00:00Z to 2016-10-15 23:59:59Z
```
Comment (morganbr) [inline src/System.Security.Cryptography.X509Certificates/tests/ChainTests.cs]: Are the issuer and root valid during both the valid and invalid times?

**Comment by morganbr** [comment]
LGTM otherwise

File: src/System.Security.Cryptography.X509Certificates/tests/ChainTests.cs (hunk 1 above)
Comment (bartonjs) [inline src/System.Security.Cryptography.X509Certificates/tests/ChainTests.cs]: This particular chain is already used for the basic sanity test, the DER bytes are at https://github.com/dotnet/corefx/blob/master/src/System.Security.Cryptography.X509Certificates/tests/TestData.cs#L944

File: src/System.Security.Cryptography.X509Certificates/tests/ChainTests.cs (hunk 2 above)
Comment (bartonjs) [inline src/System.Security.Cryptography.X509Certificates/tests/ChainTests.cs]: Yeah, Intermediate is 2013-10-30 to 2023-10-30.  Root is an optimistic 2006-11-07 to 2036-07-16.

**Comment by stephentoub** [comment]
LGTM2

**Pull request merged**